legacy/web/src/main/java/org/jboss/as/web/WebMigrateOperation.java

```diff
@@ -124,15 +130,17 @@ private WebMigrateOperation(boolean describe) {
     }
 
     static void registerOperations(ManagementResourceRegistration registry, ResourceDescriptionResolver resourceDescriptionResolver) {
-        registry.registerOperationHandler(new SimpleOperationDefinitionBuilder("migrate", resourceDescriptionResolver)
+        registry.registerOperationHandler(new SimpleOperationDefinitionBuilder(MIGRATE, resourceDescriptionResolver)
                         .setRuntimeOnly()
                         .setAccessConstraints(SensitiveTargetAccessConstraintDefinition.READ_WHOLE_CONFIG)
+                        .setReplyType(ModelType.PROPERTY).setReplyValueType(ModelType.OBJECT)
```

Comment:
Instead of this, create an AD for "migration-warnings", ModelType.LIST, value type STRING and then

.setReplyParameters(MIGRATION_WARNINGS)